job/strategies/templates/strategy.py

```diff
@@ -1,31 +1,39 @@
+from __future__ import annotations
+
 from abc import ABCMeta, abstractmethod
-from typing import Any, Dict, List
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, List
 
 import numpy as np
 import pandas as pd
 
 from db.mappings.model import ModelType
 from db.mappings.recommendation import Rec
-from sites.templates.site import Site
+
+if TYPE_CHECKING:
+    from sites.templates.site import Site
 
 
+@dataclass
```

Comment:
🔧 I'd lose this decorator. `Strategy` doesn't look like a dataclass to me since it's merely a collection of training methods/procedures, but its children, with their defined properties (e.g., training parameters) absolutely can!